### ISSUE TITLE
[Bug]: exception on one of the attachments

### ISSUE THREAD

**Issue by huyz** [rxliuli/joplin-utils#19]
### 

- [X] I checked [FAQ](https://joplin-utils.rxliuli.com/joplin-vscode-plugin/guide/faq.html)

### The tool/kit where the error occurred

joplin-blog

### Describe the bug

I'm running joplin-blog on my entire set of notes and for some reason an exception is thrown for one of the attachments:

```
$ joplin-blog wiki
    _             _ _             _     _
   (_) ___  _ __ | (_)_ __       | |__ | | ___   __ _
   | |/ _ \| '_ \| | | '_ \ _____| '_ \| |/ _ \ / _` |
   | | (_) | |_) | | | | | |_____| |_) | | (_) | (_| |
  _/ |\___/| .__/|_|_|_| |_|     |_.__/|_|\___/ \__, |
 |__/      |_|                                  |___/
  Filter to get 766 notes to be processed
  End reading note attachments and tags
  End frame initialization action
  0 cached notes have been skipped
  End of parsing the Joplin internal links and attachment resources in the notes
  End writing notes to local file
⠼ 30/59 Copying attachment resource: IMG_20130708_200823.020.jpgnode:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

[Error: ENOENT: no such file or directory, copyfile '/Users/huy/.config/joplin-desktop/resources/7de408d4ec98e0fa8314c6b092156956.' -> '/Users/huy/dev/joplin-blog-export/p/resource/7de408d4ec98e0fa8314c6b092156956.'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'copyfile',
  path: '/Users/huy/.config/joplin-desktop/resources/7de408d4ec98e0fa8314c6b092156956.',
  dest: '/Users/huy/dev/joplin-blog-export/p/resource/7de408d4ec98e0fa8314c6b092156956.'
}
error Command failed with exit code 1.
```

It seems the extension is missing from the filename.

The files do exist:
```
❯ ls /Users/huy/.config/joplin-desktop/resources/7de408d4ec98e0fa8314c6b092156956.*
/Users/huy/.config/joplin-desktop/resources/7de408d4ec98e0fa8314c6b092156956.crypted
/Users/huy/.config/joplin-desktop/resources/7de408d4ec98e0fa8314c6b092156956.jpg
```

I checked the relevant note and the markdown says:
```markdown
![IMG_20130713_134357.026.jpg](:/7de408d4ec98e0fa8314c6b092156956)
```

I don't see anything weird.

### Steps to reproduce the behavior

I don't know if you can reproduce it as the tool works for most attachments. It only fails on this one so far.

**Comment by rxliuli** [comment]
Check if the file _/Users/huy/.config/joplin-desktop/resources/7de408d4ec98e0fa8314c6b092156956.jpg_ exists

**Comment by rxliuli** [comment]
> > /Users/huy/.config/joplin-desktop/resources/7de408d4ec98e0fa8314c6b092156956.jpg
> 
> I did. I already listed the output of `ls` above in my bug description. The file exists and I'm able to open it and see the image with macOS Preview app.

Sorry, I was too careless, can you try to run it locally?

1. git clone https://github.com/rxliuli/joplin-utils.git
2. cd joplin-utils/
3. yarn && yarn setup
4. cd apps/joplin-blog/
5. edit <https://github.com/rxliuli/joplin-utils/blob/11500879dd53539aa025ae5adb46cf72142314a8/apps/joplin-blog/src/blog/Application.ts#L232>
6. yarn build
7. cd <blog-path>
8. yarn add -D <joplin-utils>/apps/joplin-blog
9. yarn joplin-blog blog

Editing Application.ts refers to printing `fileName` and `resource.file_extension`. I observed that the file name reported by the error seems to lack an extension, and the extension is obtained through joplin api. Maybe, this may be joplin itself An error

**Comment by rxliuli** [comment]
Steps 1-5 above are canceled. You can download this zip and unzip it to get the tgz file, then use `yarn add -D ./joplin-blog-0.7.0.tgz` to install and regenerate to test it?

[joplin-blog-0.7.0.zip](https://github.com/rxliuli/joplin-utils/files/7278209/joplin-blog-0.7.0.zip)

**Comment by rxliuli** [comment]
If it is wrong, please try to use the most primitive api to obtain it (replace id and token by yourself), observe the `file_extension` field in the result, if there is no value, please report to joplin official.

```sh
curl 'http://localhost:41184/resources/<id>?fields=id,title,file_extension&token=<token>'
```

> In addition, you can also try one of the simplest solutions, delete the image from this note, and then copy the picture from it and paste it again. Have a good time.

**Comment by huyz** [comment]
Yeah, the `file_extension` field is missing from the DB. These are the attachments that succeeded plus the one that failed:
<img width="1233" alt="2021-10-04 Joplin DB resources screenshot" src="https://user-images.githubusercontent.com/128394/135867973-0600f695-8444-4401-953c-9e8ec53871f5.png">

But I have a lot of attachments that have an empty `filename_extension`

**Comment by rxliuli** [comment]
> But I have a lot of attachments that have an empty `filename_extension`

Then you may need to report a bug to the official, which is obviously a mistake

**Comment by huyz** [comment]
@rxliuli Thanks for your efforts.  I've actually decided to switch to Obsidian for now.
